src/network/networkd-ndisc.c

```diff
@@ -380,15 +379,42 @@ static int ndisc_router_process_icmp6_ratelimit(Link *link, sd_ndisc_router *rt)
 
         /* Limit the maximal rates for sending ICMPv6 packets. 0 to disable any limiting, otherwise the
          * minimal space between responses in milliseconds. Default: 1000. */
-        xsprintf(buf, USEC_FMT, DIV_ROUND_UP(icmp6_ratelimit, USEC_PER_MSEC));
-
-        r = sysctl_write_ip_property(AF_INET6, NULL, "icmp/ratelimit", buf);
+        r = sysctl_write_ip_property_uint32(AF_INET6, NULL, "icmp/ratelimit",
+                (uint32_t)DIV_ROUND_UP(icmp6_ratelimit, USEC_PER_MSEC));
```

Comment:
Oh, I did similar change in #30898 in more safer way. Could you drop the change ? Sorry.

Reply:
Reverted.